node/src/components/linear_chain_sync.rs

```diff
@@ -445,12 +450,16 @@ where
                     self.block_downloaded(rng, effect_builder, block.header())
```

Comment:
I guess this is yet another place where syncing while reusing local storage would fail – we would not execute blocks that we already have in the storage, so we would not get consensus instances they would have created.

Reply:
But we do the same thing we do with incoming blocks: just call `block_downloaded`. I think that will execute them?